client/src/main/java/com/vesoft/nebula/entry/AbstractNebulaCallback.java

```diff
@@ -19,24 +19,28 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(AbstractNebulaCallback.class);
 
-    protected TBase result;
+    protected volatile TBase result;
 
     protected AtomicBoolean isReady = new AtomicBoolean(false);
 
     public Optional<?> getResult() throws InterruptedException {
         while (!checkReady()) {
             Thread.sleep(100);
         }
+        if (result == null) {
```

Comment:
Actually, `result` will filled when `doComplete` is finished. Although `result` is null, but it's obtain by the response.

Reply:
if result is null, Optional.of(result) will throw NullPointerException

Reply:
Strange ... You've met the result is null ?

Reply:
When rpc is timeout, the _onError_ is called, leaving _result_ uninitialized.